### PR TITLE
Feature/new featured navigation

### PR DESCRIPTION
**Pull request by maccesar21**
Adds the ability to specify the menu type per each column, giving the authors the ability to mix, and match the navigation structure giving emphasis to more important core products, and or featured items.

### CODE REVIEW

File: src/demo.js [hunk 1]
```diff
@@ -93,48 +241,20 @@ document.addEventListener("DOMContentLoaded", () => {
 								label: 'Free Trial',
 								href: 'https://www.esri.com/arcgis/trial'
 							}
-						],
```
[the remark this answers is not in the document]
Comment (maccesar21) [reply src/demo.js]: The solutions, and about both respectively have this feature turned on.

File: src/demo.js [hunk 2]
```diff
@@ -16,44 +16,192 @@ document.addEventListener("DOMContentLoaded", () => {
...
+										label: 'How to Buy',
+										href: 'https://www.esri.com/en-us/arcgis/analytics/overview'
+									}
+								]
 							},
 							{
-								label: 'Analytics and Data Science',
-								href: 'https://www.esri.com/en-us/arcgis/analytics/overview',
-								description: 'Advanced geospatial analysis, AI, and machine learning'
+								type: 'flyout',
```
[the remark this answers is not in the document]
Comment (maccesar21) [reply src/demo.js]: This has been updated, and removed.

**Comment by dlewis-esri** [comment]
@paulcpederson and @gschnall -- can you guys give this PR a look please?

Thanks!

**Comment by paulcpederson** [comment]
So this is a breaking change for all global-nav consumers?

**Comment by dlewis-esri** [comment]
No the existing data structure will continue to work 

New Data Structure:
header->menus[][n]->cols
https://github.com/Esri/global-nav/blob/feature/new-featured-navigation/src/demo.js#L19

Old Data structure
header->menus[][n]->menus
https://github.com/Esri/global-nav/blob/feature/new-featured-navigation/src/demo.js#L136

if the new data structure is present it will use it otherwise it will render as it has always done.

Thanks for asking - I had to make a few tweaks to ensure cols could be added without menus
@maccesar21 - please review my last 2 commits

File: src/header/js/header-menus.js [hunk 3]
```diff
@@ -265,19 +307,31 @@ export default ({variant = 'desktop'}) => {
...
-					$('a', {href: entry.href, class: `${prefix}-sublink`},
-						$('p', {class: `${prefix}-sublink--title`},
-							entry.label
-						),
-						$('p', {class: `${prefix}-sublink--description`},
-							entry.description
+			
+			if (entry.href && entry.label && entry.description) {
+				$items.push(
+					$items.push(
```
Comment (gschnall) [inline src/header/js/header-menus.js]: Why these two `items.push statements`? Do you want to get an array composed of `[item, array.length, ...]`? $items.push returns the length of the array.

File: src/header/js/header-menus.js [hunk 4]
```diff
@@ -265,19 +307,31 @@ export default ({variant = 'desktop'}) => {
...
+							$('a', {href: entry.href, class: `${prefix}-entry-sublink`},
+								$('p', {class: `${prefix}-entry-sublink--title`}, entry.label),
+								$('p', {class: `${prefix}-sublink--description`}, entry.description)
+							)
 						)
 					)
-				)
-			);
+				);
+			} else if (entry.href && entry.label) {
```
Comment (gschnall) [inline src/header/js/header-menus.js]: This `else if` statement is repetitive. You can use a ternary above to achieve the same results.

File: src/header/js/header-menus.js [hunk 5]
```diff
@@ -265,19 +307,31 @@ export default ({variant = 'desktop'}) => {
...
-						$('p', {class: `${prefix}-sublink--description`},
-							entry.description
+			
+			if (entry.href && entry.label && entry.description) {
+				$items.push(
+					$items.push(
+						$('li', {class: `${prefix}-entry--menus-subitem`},
+							$('a', {href: entry.href, class: `${prefix}-entry-sublink`},
+								$('p', {class: `${prefix}-entry-sublink--title`}, entry.label),
+								$('p', {class: `${prefix}-sublink--description`}, entry.description)
```
Comment (gschnall) [inline src/header/js/header-menus.js]: You could do something like the following instead and remove the check for `entry.description` in the if statement above.
```js
entry.description ? $('p', {class: `${prefix}-sublink--description`}, entry.description) : ""
```

File: src/header/js/header-menus.js [hunk 6]
```diff
@@ -232,30 +219,85 @@ export default ({variant = 'desktop'}) => {
...
+	function renderer(entries) {
+		const $items = [];
+
+		entries.map((entry) => {
+			if (entry.heading) {
+				$items.push(
+					$('li', {class: `${prefix}-entry--heading`},
+						$('p', {class: `${prefix}-entry--heading-label`},
+							entry.heading
+						))
```
Comment (gschnall) [inline src/header/js/header-menus.js]: These parenthesis above don't seem to be lined up correctly.

File: src/header/js/header-menus.js [hunk 7]
```diff
@@ -232,30 +219,85 @@ export default ({variant = 'desktop'}) => {
...
+			);
+		}
+	}
+	
+	function renderMulti({$subcontent, item, uuid, suuid}) {
+		const $cols = $('div', {class: `${prefix}-sublist--col-wrapper`});
+		if (item.cols) {
+			item.cols.forEach((col) => {
+				let menuType = 'standard';
+				let menuRenderer = renderer;
```
Comment (gschnall) [inline src/header/js/header-menus.js]: You can use ternaries here and remove the switch statement below.
```js
let menuType = col.type === 'structured' ? 'structured' : 'standard';
let menuRenderer = col.type === 'structured' ? renderStructuredMenu :  renderer;
```

**Comment by paulcpederson** [comment]
agree with @gschnall on pretty much every comment.

**Comment by dlewis-esri** [comment]
@gschnall @paulcpederson @maccesar21 -- We just went through and made the recommended updates

Thanks,
Daniel

**Pull request merged**